xonsh/prompt/vc.py

```diff
@@ -157,16 +145,17 @@ def current_branch():
     """
     branch = None
     cmds = builtins.__xonsh__.commands_cache
-    # check for binary only once
-    if cmds.is_empty():
-        has_git = bool(cmds.locate_binary("git", ignore_alias=True))
-        has_hg = bool(cmds.locate_binary("hg", ignore_alias=True))
-    else:
-        has_git = bool(cmds.lazy_locate_binary("git", ignore_alias=True))
-        has_hg = bool(cmds.lazy_locate_binary("hg", ignore_alias=True))
-    if has_git:
+
+    # This allows us to locate binaries after git only if necessary
+    def has(vc):
+        if cmds.is_empty():
+            return bool(cmds.locate_binary(vc, ignore_alias=True))
+        else:
+            return bool(cmds.lazy_locate_binary(vc, ignore_alias=True))
```

Comment:
This should not be defined in the closure. Please move this to the top-level, and perhaps add a leading underscore to the name